web/src/settings_users.ts

```diff
@@ -191,29 +191,49 @@ function role_selected_handler(
     widget.render();
 }
 
-function get_roles(): {
+function count_users_by_role(user_ids: number[]): Record<number, number> {
+    const role_counts: Record<number, number> = {};
+
+    for (const user_id of user_ids) {
+        const user = people.get_by_user_id(user_id);
+        const role_code = user.role;
+
+        role_counts[role_code] = (role_counts[role_code] ?? 0) + 1;
+    }
+
+    return role_counts;
+}
+
+type RoleOption = {
     unique_id: number;
     name: string;
-}[] {
+};
+
+function get_roles_with_counts(user_ids: number[]): RoleOption[] {
+    const role_counts = count_users_by_role(user_ids);
     return [
-        {unique_id: 0, name: $t({defaultMessage: "All roles"})},
+        {
+            unique_id: 0,
+            name: `${$t({defaultMessage: "All roles"})} (${user_ids.length})`, // Total count for all users in this list
```

Comment:
The parens and number should be inside the translation tag, resulting in no need for the JS string template. See
https://zulip.readthedocs.io/en/latest/translating/internationalization.html , but the core point is that different languages will want to order or punctuate these fields differently.

Also you won't need a JS template after making that change.